src/main/java/school/hei/haapi/service/WorkDocumentService.java

```diff
@@ -43,32 +42,43 @@ public WorkDocument uploadStudentWorkFile(
       Instant creationDatetime,
       Instant commitmentBegin,
       Instant commitmentEnd,
-      WorkStudyStatus studentWorkStatus,
       MultipartFile workFile) {
     User student = userService.findById(studentId);
 
-    student.setWorkStatus(
-        defineStudentWorkStatus(commitmentBegin, commitmentEnd, studentWorkStatus));
-    student.setCommitmentBeginDate(commitmentBegin);
-    userRepository.save(student);
-
     return fileInfoService.uploadFile(
         student, filename, creationDatetime, commitmentBegin, commitmentEnd, workFile);
   }
 
-  private WorkStudyStatus defineStudentWorkStatus(
-      Instant commitmentBegin, Instant commitmentEnd, WorkStudyStatus studentWorkStatus) {
-    Instant now = Instant.now();
+  public Optional<WorkDocument> findLastWorkDocumentByStudentId(String studentId) {
+    return workDocumentRepository.findTopByStudentIdOrderByCreationDatetimeDesc(studentId);
+  }
 
-    if (studentWorkStatus != null) {
-      return studentWorkStatus;
+  public Instant defineStudentCommitmentBegin(Optional<WorkDocument> workDocument) {
+    if (!workDocument.isPresent()) {
+      return null;
+    }
+    return workDocument.get().getCommitmentBegin();
+  }
+
+  public WorkStudyStatus defineStudentWorkStatusFromWorkDocumentDetails(
+      Optional<WorkDocument> workDocument) {
+    if (!workDocument.isPresent()) {
+      return NOT_WORKING;
+    }
+
+    Instant now = Instant.now();
+    WorkDocument domainWorkDocument = workDocument.get();
+    if (domainWorkDocument.getCommitmentBegin() != null) {
+      return WORKING;
     }
-    if (now.isBefore(commitmentBegin)) {
+    if (domainWorkDocument.getCommitmentBegin() != null
+        && now.isBefore(domainWorkDocument.getCommitmentBegin())) {
       return WILL_BE_WORKING;
     }
-    if (now.isAfter(commitmentEnd)) {
+    if (domainWorkDocument.getCommitmentEnd() != null
+        && now.isAfter(domainWorkDocument.getCommitmentEnd())) {
       return HAVE_BEEN_WORKING;
     }
-    return WORKING;
+    return null;
```

Comment:
wrap in optionalWorkdocument.map(action).orElse(NOT_WORKING).
null values aren't to be avoided as often as possible